replica/compaction_group.hh

```diff
@@ -121,6 +123,61 @@ public:
     uint64_t total_disk_space_used() const noexcept;
 
     compaction::table_state& as_table_state() const noexcept;
+
+    class sstable_set;
```

Comment:
Let's move it out of compaction_group. First, it's not an sstable_set for a compaction group, but for all compaction groups. Second, nested classes are a little impractical.

Reply:
(for forward declarations; though that's not a real problem here since it's hidden by a factory function)